### PR TITLE
Feature segment

### PR DESCRIPTION
**Pull request by rakshasa**
Adds support for virtual segments.

### CODE REVIEW

**Comment by rakshasa** [comment]
All green.

File: integration_test/lib/vnspec/vm.rb [hunk 1]
```diff
@@ -248,6 +256,10 @@ def ipv4_address
         end
       end
 
+      def change_ipv4_address(new_address, new_prefix = 24)
```
Comment (akry) [inline integration_test/lib/vnspec/vm.rb]: Is this method really necessary? Nobody calls it.

File: vnet/lib/vnet/core/services/dhcp.rb [hunk 2]
```diff
@@ -188,6 +201,10 @@ def find_client_infos(port_number, server_mac_info, server_ipv4_info)
     def find_client_interface(port_number, server_mac_info, server_ipv4_info)
       interface = @dp_info.interface_manager.detect(port_number: port_number)
 
+      debug "ZZZZZZZZZZZZZZZZZZZZ #{server_mac_info.inspect}"
+      debug "ZZZZZZZZZZZZZZZZZZZZ #{server_ipv4_info.inspect}"
+      debug "ZZZZZZZZZZZZZZZZZZZZ #{interface.inspect}"
+
```
Comment (akry) [inline vnet/lib/vnet/core/services/dhcp.rb]: Could you delete these debug messages if you don't need them anymore?

**Comment by akry** [comment]
Overall there are 19 `|| return` statements without any error message in this PR. This might have a huge impact on future usage so friendly error messages are very welcome.

**Comment by rakshasa** [comment]
Those are copy-n-paste'ed code, will need to fix the other code too.

File: integration_test/lib/vnspec/vm.rb (hunk 1 above)
Comment (rakshasa) [inline integration_test/lib/vnspec/vm.rb]: It's code that will be used in the simple_seg test once I've refactored the dhcp/static address configuration for CI interfaces.

Comment (akry) [inline integration_test/lib/vnspec/vm.rb]: If you need this method it's ok. But it should be included in another PR I think.

**Comment by Metallion** [comment]
I will merge as soon as the remaining `|| return` statements have been replaced by proper error messages.

**Comment by rakshasa** [comment]
All errors handling in the branch has been changed to use get_params. The places that still use '|| return' are valid code paths.

**Comment by Metallion** [comment]
I know I said I'd merge already but I just noticed there are no tests for this yet. It looks like there's currently only a partially written test scenario that's not being executed yet. https://github.com/axsh/openvnet/blob/25b3c84c0bc510a5e7c3c64bb5ec15ce4556aa94/integration_test/lib/vnspec/spec/simple_seg_spec.rb

Please finish up this test scenario and add it to the executed tests first.

File: integration_test/lib/vnspec/spec/simple_seg_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+require_relative "spec_helper"
+
+describe "simple_seg" do
```
Comment (Metallion) [inline integration_test/lib/vnspec/spec/simple_seg_spec.rb]: I think this abbreviation is quite obvious enough. When other developers look at the integration test, they won't immediately know that seg means segment. I think it's better to write "simple_segment" without abbreviating.

**Comment by rakshasa** [comment]
http://ci.wakame.jp/job/openvnet.optimized.integration-test/249/

All green. More complex segment test (including vrouters and such) require features not part of this PR.

**Comment by Metallion** [comment]
The integration test looks good but still needs to run rspec since there have been new commits since last time. I'll quickly run it.
